contracts/contracts/pausing/Pauser.sol

```diff
@@ -0,0 +1,232 @@
+        external
+        onlyGovernor
+        initializer
+    {
+        setPausable(_pausable);
+        setExpiryDuration(_expiryDuration);
+    }
+
+    function addToWhitelist(address _account)
+        external
+        onlyGovernorOrStrategist
+    {
+        whitelist[_account] = true;
+        emit Whitelisted(_account);
+    }
+
+    function removeFromWhitelist(address _account)
+        external
+        onlyGovernorOrStrategist
+    {
+        whitelist[_account] = false;
+        emit DeWhitelisted(_account);
+    }
+
+    /**
+     * @dev Pauses the managed contract for the duration of `expiryDuration`
+     * The pause action can be canceled by any user when the temp-pause expires.
+     * The contract MUST be in an UNPAUSED state.
+     * The function is one-time-use (The contract cannot be temp-paused by the same account twice).
+     */
+    function tempPause() external onlyWhitelistedOrGovernorOrStrategist {
+        // Cannot pause if already paused
+        require(pauseState == UNPAUSED, "Contract is not unpaused");
+
+        if (!_isGovernorOrStrategist()) {
+            // Cannot initiate a temp-pause if done before (one-time-use)
+            require(
+                pauseExpiry[msg.sender] == 0,
+                "Caller has already initiated a temp pause"
+            );
+        }
+
+        pauseState = TEMP_PAUSE;
+        currentExpiry = block.timestamp + expiryDuration;
+        pauseExpiry[msg.sender] = currentExpiry;
+        emit TempPaused(currentExpiry);
+
+        // Execute the pause action on the contract
+        Pausable(pausable).pause();
+    }
+
+    /**
+     * @dev Confirm a temporary pause on the contract
+     * The contract MUST be in a TEMP_PAUSE state.
+     */
+    function confirmPause() external onlyGovernorOrStrategist {
+        // The contract MUST be in a TEMP_PAUSE state
+        require(pauseState == TEMP_PAUSE, "The contract is not temp-paused");
+
+        // currentExpiry MUST never be zero when pauseState = TEMP_PAUSE
+        assert(currentExpiry != 0);
+
+        pauseState = PAUSED;
+        currentExpiry = 0;
+        emit Paused();
+    }
+
+    /**
+     * @dev Cancel the temporary pause on the contract after expiry.
+     * The contract MUST be in a TEMP_PAUSE state.
+     */
+    function cancelTempPause() external onlyWhitelistedOrGovernorOrStrategist {
+        // Cannot cancel if not temp-paused
+        require(pauseState == TEMP_PAUSE, "The contract is not temp-paused");
+
+        // Cannot cancel before expiry
+        require(
+            block.timestamp > currentExpiry,
+            "The current pause is not expired"
+        );
+
+        pauseState = UNPAUSED;
+        currentExpiry = 0;
+        emit Unpaused();
+
+        // Execute the unpausing action on the contract
+        Pausable(pausable).unpause();
+    }
+
+    /**
+     * @dev Pauses the managed contract indefinitely
+     */
+    function pause() external onlyGovernorOrStrategist {
```

Comment:
When someone needs to call pause on this contract, it's going to be very urgent. Because of that, we want to make it as simple and hard to mess up as possible.

So in this case, I think we should collapse the the `tempPause`, `confirmPause`, `cancelTempPause`, `pause`, and `unpause` methods down to just `pause` and `unpause`. This way it's always clear to the user under stress what should be used. They just need to know what state they want the contract to be in, and don't have to worry about the state that the contract is currently in.

`Pause()` moving to a full pause or a temp pause depends on who is calling it.

<img width="489" alt="image" src="https://user-images.githubusercontent.com/837/169600132-e9c79fe3-841e-4b98-b932-6d51caae53ba.png">

<img width="484" alt="image" src="https://user-images.githubusercontent.com/837/169600152-7ba1c9e2-363d-4596-9a08-64b0567311d8.png">

<img width="499" alt="image" src="https://user-images.githubusercontent.com/837/169600177-d49e07e7-1ca4-45a8-9b2d-de68783ae5c8.png">

Reply:
Note two differences in this diagram from the current code:


1. Any member of the public can unpause, once the expiration is past. Not just privileged users. The intent is to not have a way that the contract funds could be permanently lost if there were no strategist and a whitelisted user was malicious.

2. Whitelisted users cannot unpause from a temp pause until the expiration time is up, at which point they are like anyone else. This gives us a guaranteed time to react and investigate.

Reply:
Good idea. I think it also makes it easier to maintain. Thanks for the review.